filecache/__init__.py

```diff
@@ -65,29 +64,40 @@
         with fc.open('gs://my-writable-bucket/output.txt', 'r') as fp:
             print(fp.read())
 
-A :class:`FileCachePrefix` instance can be used to encapsulate the storage prefix string,
-as well as any subdirectories, plus various arguments such as `anonymous` and `time_out`
-that can be specified to each `exists`, `retrieve`, or `upload` method. Thus using one
-of these instances can simplify the use of a :class:`FileCache` by allowing the user to
-only specify the relative part of the path to be operated on, and to not specify various
-other parameters at each method call site.
+The :class:`FCPath` class is a reimplementation of the Python `Path` class to support
+remote acess using an associated :class:`FileCache`. Like `Path`, an :class:`FCPath`
+instance can contain any part of a URI, but only an absolute URI can be used when actually
+accessing the file specified by the :class:`FCPath`. In addition, an :class:`FCPath` can
+encapsulate various arguments such as `anonymous` and `time_out` so that they do not need
+to be specified to each access method. Thus, using this class can simplify the use of a
+:class:`FileCache` by allowing the user to operate on paths using the simpler syntax
+provided by `Path`, and to not specify various other parameters at each method call site.
+If an :class:`FCPath` instance is created without an explicitly-associated
+:class:`FileCache`, then the default `FileCache()` is used, which specifies a shared cache
+named `"global"` that will persist after the program exits.
 
 Compare this example to the one above::
 
-    from filecache import FileCache
+    from filecache import FileCache, FCPath
+    # Create a cache with a unique name that will be deleted on exit
     with FileCache(None) as fc:  # Use as context manager
         # Use GS by specifying the bucket name and one directory level
-        pfx1 = fc.new_prefix('gs://rms-filecache-tests/subdir1', anonymous=True)
+        p1 = fc.new_path('gs://rms-filecache-tests/subdir1', anonymous=True)
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider impact of method name change</b></div>
<div id="fix">

Consider if renaming `new_prefix` to `new_path` could potentially break backward compatibility for existing code that relies on the old method name. A similar issue was also found in filecache/__init__.py (line 178).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -80,1 +85,6 @@
 -        pfx1 = fc.new_prefix('gs://rms-filecache-tests/subdir1', anonymous=True)
 +        # Keep old method but mark as deprecated
 +        def new_prefix(self, *args, **kwargs):
 +            import warnings
 +            warnings.warn('new_prefix is deprecated, use new_path instead', DeprecationWarning)
 +            return self.new_path(*args, **kwargs)
 +        p1 = fc.new_path('gs://rms-filecache-tests/subdir1', anonymous=True)
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/SETI/rms-filecache/pull/24#issuecomment-2495681924>#ad05ee</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

Reply:
Good suggestion.